commercial-schedules/src/main/java/org/dcsa/conformance/standards/cs/model/CsDateUtils.java

```diff
@@ -0,0 +1,122 @@
+    String date =
+        extractValue(queryParams, "date")
+            .map(dateToProcess -> processDate(dateToProcess, "", "date"))
+            .orElseGet(() -> CsDateUtils.DATE_TIME_FORMATTER.format(ZonedDateTime.now()));
+    entryMap.put("DATE", date);
+  }
+
+  private static String getProcessedDate(
+      Map<String, ? extends Collection<String>> queryParams,
+      String startDateParam,
+      String endDateParam,
+      String mapKey) {
+    Optional<String> startDateOpt = extractValue(queryParams, startDateParam);
+    Optional<String> endDateOpt = extractValue(queryParams, endDateParam);
+
+    if (startDateOpt.isPresent() || endDateOpt.isPresent()) {
+      String startDate = startDateOpt.orElse(null);
+      String endDate = endDateOpt.orElse(null);
+      return "ARRIVAL_DATE".equals(mapKey)
+          ? getArrivalDateTime(startDate, endDate)
+          : getDepartureDateTime(startDate, endDate);
+    } else {
+      return CsDateUtils.DATE_TIME_FORMATTER.format(ZonedDateTime.now());
+    }
+  }
+
+  private static Optional<String> extractValue(
+      Map<String, ? extends Collection<String>> queryParams, String key) {
+    return Optional.ofNullable(queryParams.get(key))
+        .flatMap(collection -> collection.stream().findFirst());
+  }
+
+  private static String getArrivalDateTime(String arrivalStartDate, String arrivalEndDate) {
+    if (arrivalStartDate != null && arrivalEndDate != null) {
+      return processDate(arrivalStartDate, arrivalEndDate, "range");
+    }
+    if (arrivalStartDate != null) {
+      return processDate(arrivalStartDate, "", "startDate");
+    }
+    if (arrivalEndDate != null) {
+      return processDate("", arrivalEndDate, "endDate");
+    }
+    return "";
+  }
+
+  private static String getDepartureDateTime(String departureStartDate, String departureEndDate) {
+    if (departureStartDate != null && departureEndDate != null) {
+      return processDate(departureStartDate, departureEndDate, "range");
+    }
+    if (departureStartDate != null) {
+      return processDate(departureStartDate, "", "startDate");
+    }
+    if (departureEndDate != null) {
+      return processDate("", departureEndDate, "endDate");
+    }
+    return "";
+  }
+
+  private static String processDate(String startDate, String endDate, String type) {
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    if (type.equals("startDate")) {
+      LocalDate date = LocalDate.parse(startDate, formatter).plusWeeks(1);
+      return convertDateToDateTime(date);
+    }
+    if (type.equals("endDate")) {
+      LocalDate date = LocalDate.parse(endDate, formatter).minusWeeks(1);
+      return convertDateToDateTime(date);
+    }
+    if (type.equals("range")) {
+      LocalDate date = LocalDate.parse(startDate, formatter).plusWeeks(1);
+      return convertDateToDateTime(date);
+    }
+    if (type.equals("date")) {
```

Comment:
With IntelliJ, you can easily make this into a switch statement, which is easier to read and maintain.

Reply:
Thanks for the improvements @palatsangeetha. You could even make it more concise, by doing this idea:
```
switch (type) {
      case "startDate", "range":
        return convertDateToDateTime(LocalDate.parse(startDate, formatter).plusWeeks(1));
```
Than you don't need the breaks and the end return at line 111. 😉

Reply:
Yes, that's the enhanced switch feature, introduced in Java 17, even more clean, thanks GJ.